src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGenYul.scala

```diff
@@ -132,16 +150,217 @@ object CodeGenYul extends CodeGenerator {
+                        Seq(ExpressionStatement(expr))
+                    case _ =>
+                        assert(false, "TODO")
+                        Seq()
+                }
+
+            case _ =>
+                assert(false, "TODO")
+                Seq()
+
+        }
+    }
+
+    def translateExpr(e: Expression): Seq[YulStatement] = {
+        e match {
+            case ReferenceIdentifier(x) =>
+                val idx = temp_symbol_table(x)
+                val expr = FunctionCall(Identifier("sload"), Seq(Literal(LiteralKind.number, idx.toString(), "int")))
+                Seq(ExpressionStatement(expr))
+            case _ => Seq() // TODO unimplemented
+        }
     }
 }
 
 // temporary function, not designed for a full recursive walk through of the object
-class Scope(val mainContractName: String) {
-    val creationObject:String = mainContractName
-    val runtimeObject:String = mainContractName + "_deployed"
-}
\ No newline at end of file
+class ObjScope(obj: YulObject) {
+    class Func(val code: String){}
+    class Case(val hash: String){}
+    class Call(val call: String){}
+
+    // TODO unimplemented; hardcode to uint256 for now
+    def mapObsTypeToABI(ntype: String): String = {
+        "uint256"
+    }
+
+    // TODO unimplemented; hardcode for now
+    def keccak256(s: String): String = {
+        "0x70a08231"
+    }
+
+    def hashFunction(f: FunctionDefinition): String = {
+        var strRep: String = f.name + "("
+        for (p <- f.parameters){
+            strRep = strRep + mapObsTypeToABI(p.ntype)
+        }
+        strRep = strRep + ")"
+        keccak256(strRep)
+        // TODO truncate and keep the first 4 bytes
+    }
+
+    val mainContractName: String = obj.name
+    val creationObject: String = mainContractName
+    val runtimeObject: String = mainContractName + "_deployed"
+    var runtimeFunctionArray: Array[Func] = Array[Func]()
+    var deployFunctionArray: Array[Func] = Array[Func]()
+    var dispatch = false
+    var dispatchArray: Array[Case] = Array[Case]()
+    var deployCall: Array[Call] = Array[Call]()
+    var memoryInitRuntime: String = ""
+
+    for (s <- obj.code.block.statements) {
+        s match {
+            case f: FunctionDefinition => deployFunctionArray = deployFunctionArray :+ new Func(yulString.yulFunctionDefString(f))
+            case e: ExpressionStatement =>
+                e.expression match {
+                    case f: FunctionCall => deployCall = deployCall :+ new Call(yulString.yulFunctionCallString(f))
+                    case _ => () // TODO unimplemented
+                }
+            case _ => () // TODO unimplemented
+        }
+    }
+
+    for (sub <- obj.subObjects) { // TODO separate runtime object out as a module
+        for (s <- sub.code.block.statements) { // temporary fix due to issue above
```

Comment:
Which issue?

Reply:
The issue is there will be an unintended object called "Contract" from the program. (see line 75). I don't know why that is the case

Reply:
Okay. Please add a comment?